src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [selectUserId, setselectUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [listTodos, setListTodos] = useState<Todo[]>(todos);
+
+  const [hasNameError, setHasNameError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const valideData = `zxcvbnmasdfghjklqwertyuiopZXCVBNMASDFGHJKLQWERTY
+  UIOPячсмитьбюЯЧСМИТЬБЮфывапролджэФЫВАПРОЛДЖЭйцукенгшщзхъЙЦУКЕНГШЩЗХёЁ123456789`;
```

Comment:
nope, just check if string empty or not, you don't need this variable

Reply:
point 5